src/ott/solvers/linear/univariate.py

```diff
@@ -12,134 +12,247 @@
+  slices. The distance is computed using the analytical formula by default,
+  which involves sorting each of the slices independently. The optimal transport
+  matrices are also outputted when possible (described in sparse form, i.e.
+  pairs of indices and mass transferred between those indices).
 
-  .. warning::
-    This solver assumes uniform marginals, a non-uniform marginal solver
-    is coming soon.
+  When weights ``a`` and ``b`` are uniform, and ``n=m``, the computation only
+  involves comparing sorted entries per slice, and ``d`` assignments are given.
 
-  Computes the 1-Dimensional optimal transport distance between two histograms.
+  The user may also supply a ``num_subsamples`` parameter to extract as many
+  points from the original point cloud, sampled with probability masses ``a``
+  and ``b``. This then simply applied the method above to the subsamples, to
+  output ``d`` costs, but assignments are not provided.
+
+  When the problem is not uniform or not of equal size, the method defaults to
+  an inversion of the CDF, and outputs both costs and transport matrix in sparse
+  form.
+
+  When a ``quantiles`` argument is passed, either specifying explicit quantiles
+  or a grid of quantiles, the distance is evaluated by comparing the quantiles
+  of the two point clouds on each slice. The OT costs are returned but
+  assignments are not provided.
 
   Args:
-    sort_fn: The sorting function. If :obj:`None`,
-      use :func:`hard-sorting <jax.numpy.sort>`.
-    cost_fn: The cost function for transport. If :obj:`None`, defaults to
-      :class:`PNormP(2) <ott.geometry.costs.PNormP>`.
-    method: The method used for computing the distance on the line. Options
-      currently supported are:
-
-      - `'subsample'` - Take a stratified sub-sample of the distances.
-      - `'quantile'` - Take equally spaced quantiles of the distances.
-      - `'equal'` - No subsampling is performed, requires distributions to have
-        the same number of points.
-      - `'wasserstein'` - Compute the distance using the explicit solution
-        involving inverse CDFs.
-
-    n_subsamples: The number of samples to draw for the "quantile" or
-      "subsample" methods.
+    num_subsamples: option to reduce the size of inputs by doing random
+      subsampling, taken into account marginal probabilities.
+    quantiles: When a vector or a number of quantiles is passed, the distance
+      is computed by evaluating the cost function on the sectional (one for each
+      dimension) quantiles of the two point cloud distributions described in the
+      problem.
   """
 
   def __init__(
       self,
-      sort_fn: Optional[Callable[[jnp.ndarray], jnp.ndarray]] = None,
-      cost_fn: Optional[costs.CostFn] = None,
-      method: Literal["subsample", "quantile", "wasserstein",
-                      "equal"] = "subsample",
-      n_subsamples: int = 100,
+      num_subsamples: Optional[int] = None,
```

Comment:
Am wondering whether we should add option for this to be a fraction, i.e., `num_subsamples: Optional[Union[int, float]]`. Wdyt?

Reply:
i think it's likely ``num_subsamples`` will be static in applications to enable more efficient jitting, so I'd rather let that one static